utils/config_test.go

```diff
@@ -37,6 +37,41 @@ func TestReadConfig(t *testing.T) {
 	require.EqualError(t, err, "parsing error at line 3, character 5: invalid character 'm' looking for beginning of object key string")
 }
 
+func TestReadConfig_PluginSettings(t *testing.T) {
+	TranslationsPreInit()
+
+	config, _, err := ReadConfig(bytes.NewReader([]byte(`{
+		"PluginSettings": {
+			"Directory": "/temp/mattermost-plugins",
+			"Plugins": {
+				"com.example.plugin": {
+					"number": 1,
```

Comment:
Given the weirdness of the original issue, can you add another configuration that, itself, contains a period and maybe has a more complex type? e.g. 
```go
"struct.important": {
    "nested.number": 1, 
    "nested.string": "abc", 
    "nested.boolean": false
}
```

Just to ensure we don't have any more weirdness going forward :)